connect/mirror/src/test/java/org/apache/kafka/connect/mirror/MirrorConnectorsIntegrationTest.java

```diff
@@ -150,23 +152,73 @@ public void setup() throws IOException {
         mm2Props.put("backup.bootstrap.servers", backup.kafka().bootstrapServers());
         mm2Config = new MirrorMakerConfig(mm2Props);
 
+        // we wait for the connector and tasks to come up for each connector, so that when we do the
+        // actual testing, we are certain that the tasks are up and running; this will prevent
+        // flaky tests where the connector and tasks didn't start up in time for the tests to be
+        // run
         backup.configureConnector("MirrorSourceConnector", mm2Config.connectorBaseConfig(new SourceAndTarget("primary", "backup"),
             MirrorSourceConnector.class));
+        waitForConnectorToBeUp(backup, "MirrorSourceConnector");
+        waitForTasksToBeUp(backup, "MirrorSourceConnector");
 
         backup.configureConnector("MirrorCheckpointConnector", mm2Config.connectorBaseConfig(new SourceAndTarget("primary", "backup"),
             MirrorCheckpointConnector.class));
+        waitForConnectorToBeUp(backup, "MirrorCheckpointConnector");
+        waitForTasksToBeUp(backup, "MirrorCheckpointConnector");
 
         backup.configureConnector("MirrorHeartbeatConnector", mm2Config.connectorBaseConfig(new SourceAndTarget("primary", "backup"),
             MirrorHeartbeatConnector.class));
+        waitForConnectorToBeUp(backup, "MirrorHeartbeatConnector");
+        waitForTasksToBeUp(backup, "MirrorHeartbeatConnector");
 
         primary.configureConnector("MirrorSourceConnector", mm2Config.connectorBaseConfig(new SourceAndTarget("backup", "primary"),
             MirrorSourceConnector.class));
+        waitForConnectorToBeUp(primary, "MirrorSourceConnector");
+        waitForTasksToBeUp(primary, "MirrorSourceConnector");
 
         primary.configureConnector("MirrorCheckpointConnector", mm2Config.connectorBaseConfig(new SourceAndTarget("backup", "primary"),
             MirrorCheckpointConnector.class));
+        waitForConnectorToBeUp(primary, "MirrorCheckpointConnector");
+        waitForTasksToBeUp(primary, "MirrorCheckpointConnector");
 
         primary.configureConnector("MirrorHeartbeatConnector", mm2Config.connectorBaseConfig(new SourceAndTarget("backup", "primary"),
             MirrorHeartbeatConnector.class));
+        waitForConnectorToBeUp(primary, "MirrorHeartbeatConnector");
+        waitForTasksToBeUp(primary, "MirrorHeartbeatConnector");
+
+    }
+
+    private void waitForConnectorToBeUp(EmbeddedConnectCluster connectCluster, String connName)
+        throws InterruptedException {
+        Collection<String> connectors = connectCluster.connectors();
+        while (!connectors.contains(connName)) {
+            // this means the connector hasn't been registered yet in the REST endpoint, so we
+            // sleep for 3 seconds and try again
+            Thread.sleep(3_000);
```

Comment:
I think this could be shorter. We don't want to add unnecessarily to build time. maybe 100ms?

Reply:
Sure we can definitely do that. After running the test a bunch of times, I found that on average it took about 3 seconds for the tasks to come up, so I tried to optimize for CPU usage (by not constantly hitting the REST endpoint until the tasks were up) instead of build time, but in this case if we don't want to add unnecessarily to build time we can definitely shorten it. With respect to how long we want to sleep, do you think 100ms is good enough to not put a lot of load on the REST endpoint or do you think we could go a bit higher?